From 7190e8f7f3f1979a3f4c886a8fc000483b647598 Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Thu, 18 Oct 2018 14:44:33 -0400
Subject: [PATCH] OS-7310 vminfod event stream processes events after stop
 called Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon
 <josh@wilsdon.ca>

---
 src/vm/node_modules/vminfod/client.js | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/vm/node_modules/vminfod/client.js b/src/vm/node_modules/vminfod/client.js
index aaac58a7..605cca78 100644
--- a/src/vm/node_modules/vminfod/client.js
+++ b/src/vm/node_modules/vminfod/client.js
@@ -482,6 +482,14 @@ VminfodEventStream.prototype._transform =
     var parsed;
     var vms;
 
+    if (!self.vs_req) {
+        // we are stopped - ignore any incoming data
+        self.vs_logger.trace({line: line},
+            'stream stopped - ignoring line from vminfod');
+        cb();
+        return;
+    }
+
     try {
         ev = JSON.parse(line);
         ev.date = new Date(ev.date);
-- 
2.21.0

